guides/keras_cv/writing_custom_image_augmentations_with_baseimageaugmentationlayer.py

```diff
@@ -0,0 +1,435 @@
+"""
+Title: Custom Image Augmentations with BaseImageAugmentationLayer
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/04/26
+Last modified: 2022/04/26
+Description: Use BaseImageAugmentationLayer to implement custom data augmentations.
+"""
+
+"""
+## Overview
+Data augmentation is an integral part of training any robust computer vision model.
+While KerasCV offers a plethora of prebuild high quality data augmentation techniques,
+you may still want to implement your own custom technique.
+KerasCV offers a helpful base class for writing data augmentation layers:
+`BaseImageAugmentationLayer`.
+Any augmentation layer built with `BaseImageAugmentationLayer` will automatically be
+compatible with the KerasCV `RandomAugmentationPipeline` class.
+
+This guide will show you how to implement your own custom augmentation layers using
+`BaseImageAugmentationLayer`.  As an example, we will implement a layer that tints all
+images blue.
+It is recommended that you read (TODO LINK TO OTHER TUTORIAL ONCE POSTED) as a precursor
+to this tutorial.
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+import keras_cv
+from tensorflow.keras import layers
+from keras_cv.utils import preprocessing as preprocessing_utils
+from keras_cv.layers import BaseImageAugmentationLayer
+from keras_cv import bounding_boxes
+import matplotlib.pyplot as plt
```

Comment:
For good readability, it would be nice to sort non-ml, ml packages. For example

```python
import matplotlib.pyplot as plt

import keras_cv
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
from keras_cv import bounding_boxes
from keras_cv.layers import BaseImageAugmentationLayer
from keras_cv.utils import preprocessing as preprocessing_utils
```